### PR TITLE
Rework buildmw() to use named arguments

### PR DESCRIPTION
**Pull request by ix5**
## build_packages.sh
Allow supplying multiple `.spec` files

## `buildmw()`: Use named parameters

Instead of relying on positional arguments, allow the following:
- `-u` for git URL
- `-b` for branch name
- `-s` for spec file. Can be supplied multiple times

## mw: Use named opts to call buildmw()

Using named arguments for calling `buildmw()` makes the whole thing more readable.
Use full URLs as well for better readability.

## `build()`: Consider all supplied spec files

Using `$@` instead of `$1` means all arguments are iterated over when looping `$SPECS`

### CODE REVIEW

File: helpers/build_packages.sh [hunk 1]
```diff
@@ -79,7 +80,7 @@ while true; do
           shift;;
       -s|--spec) BUILDSPEC=1
           case "$2" in
-              *) BUILDSPEC_FILE=$2;;
+              *) BUILDSPEC_FILE+=("$2");;
```
[the remark this answers is not in the document]
Comment (sledges) [reply helpers/build_packages.sh]: Thank you for your contribution. In the past we never had a case of building multiple .spec files. However now with the Sailfish X Android 8 adaptations we will need a mechanism to do so, and that is exactly what you have just implemented, cheers!

The historical reason why all possible .spec files used to be built by default (https://github.com/mer-hybris/droid-hal-device/blob/8bb1f91a7f0bee0fd165d304f36ffd7e1d5b7a0c/helpers/util.sh#L230) remains hidden with the community person who initially helped writing these build scripts.

File: helpers/build_packages.sh [hunk 2]
```diff
@@ -142,38 +144,51 @@ if [ "$BUILDMW" == "1" ]; then
...
             # pull device's bluez4 configs correctly
             sb2 -t $VENDOR-$DEVICE-$PORT_ARCH -m sdk-install -R zypper remove bluez-configs-mer
         fi
     else
-        buildmw $BUILDMW_REPO $BUILDSPEC_FILE || die
+        if [[ -z "$BUILDSPEC_FILE" ]]; then
+            buildmw -u $BUILDMW_REPO || die
+        else
+            # Supply all given spec files from $BUILDSPEC_FILE array prefixed with "-s"
+            buildmw -u $BUILDMW_REPO "${BUILDSPEC_FILE[@]/#/-s }" || die
```
[the remark this answers is not in the document]
Comment (sledges) [reply helpers/build_packages.sh]: :+1:

File: helpers/fix_dangling_adaptation-community_repo.sh [hunk 3]
```diff
@@ -16,12 +16,12 @@ if [ -n "$(grep '%define community_adaptation' $ANDROID_ROOT/hybris/droid-config
         echo "adaptation-community has been found pointing to an OBS repo, let's purge it for local builds"
         sb2 -t $VENDOR-$DEVICE-$PORT_ARCH -m sdk-install -R rpm -e --nodeps community-adaptation-devel
         BUILDALL=y
-        buildmw https://github.com/mer-hybris/community-adaptation.git rpm/community-adaptation-localbuild.spec || die
+        buildmw -u "https://github.com/mer-hybris/community-adaptation.git" -s rpm/community-adaptation-localbuild.spec || die
```
Comment (sledges) [inline helpers/fix_dangling_adaptation-community_repo.sh]: This file is an old relic, and will be removed altogether in a separate commit.

File: helpers/build_packages.sh [hunk 4]
```diff
@@ -187,7 +202,11 @@ if [ "$BUILDPKG" == "1" ]; then
     if [ -z $BUILDPKG_PATH ]; then
        echo "--build requires an argument (path to package)"
     else
-        buildpkg $BUILDPKG_PATH $BUILDSPEC_FILE
+        if [[ -z "$BUILDSPEC_FILE" ]]; then
```
Comment (sledges) [inline helpers/build_packages.sh]: I don't see the need to check if it's empty...

File: helpers/build_packages.sh [hunk 5]
```diff
@@ -187,7 +202,11 @@ if [ "$BUILDPKG" == "1" ]; then
...
+            buildpkg $BUILDPKG_PATH
+        else
+            buildpkg $BUILDPKG_PATH "$BUILDSPEC_FILE"
```
Comment (sledges) [inline helpers/build_packages.sh]: ...as this should suffice (previously it would pass only the first argument):
`buildpkg $BUILDPKG_PATH ${BUILDSPEC_FILE[@]}`

This https://github.com/mer-hybris/droid-hal-device/pull/205/files?utf8=%E2%9C%93&diff=unified#diff-748fa8f198703f775c5df5dc1fa219ccR295 then should be corrected as:
```
shift
build $@
```

Comment (ix5) [reply helpers/build_packages.sh]: Good catch, I've incorporated your suggestion now.

File: helpers/build_packages.sh (hunk 4 above)
Comment (ix5) [reply helpers/build_packages.sh]: Dropped

File: helpers/fix_dangling_adaptation-community_repo.sh (hunk 3 above)
Comment (ix5) [reply helpers/fix_dangling_adaptation-community_repo.sh]: Well I don't want to change any real behaviour in this commit. If you push any change I will rebase and incorporate it, but I would like to keep this PR clean.

Comment (sledges) [reply helpers/fix_dangling_adaptation-community_repo.sh]: Just pushed to master, let's keep this PR also minimal :)

Comment (ix5) [reply helpers/fix_dangling_adaptation-community_repo.sh]: Rebased and deleted.

**Comment by ix5** [comment]
@sledges If you could do a test run on your machine as well to make sure I haven't overlooked any grave errors, I think this should be ready for merge.

**Comment by sledges** [comment]
I've already tested it; please squash all commits in one so it reflects one logical change, avoiding breakage if accidentally only one of the commits gets reverted.

**Comment by ix5** [comment]
Squashed and force-pushed

**Comment by sledges** [comment]
Many thanks. Since you didn't know that we never built multiple .spec files, your commit message describes of what has been done to fix/implement this functionality.

Could you please instead word it in such a way so that readers of the commit list would instantly know that this functionality has now been introduced, e.g. allow building multiple .spec files.

A documentation update would also be appreciated to (e.g. add "can be repeated to build multiple .spec files"):
https://github.com/mer-hybris/droid-hal-device/blob/8bb1f91a7f0bee0fd165d304f36ffd7e1d5b7a0c/helpers/build_packages.sh#L44

I have added the following commit to fix the build of Xperia XA2:
https://github.com/sledges/droid-hal-device/commit/c8b9c559e9a67ddfab9e8a8e53e0d6508caeca69

Full instructions how to build XA2 (including utilising multiple .spec builds) will be made available after this PR is merged (similar to https://sailfishos.org/wiki/Sailfish_X_Build_and_Flash which was only for the Xperia X series)

**Comment by ix5** [comment]
> Many thanks. Since you didn't know that we never built multiple .spec files, your commit message describes of what has been done to fix/implement this functionality.
> Could you please instead word it in such a way so that readers of the commit list would instantly know that this functionality has now been introduced, e.g. allow building multiple .spec files.

Done, new commit msg is `Rework buildmw to allow multiple spec files`

> A documentation update would also be appreciated to (e.g. add "can be repeated to build multiple .spec files"):

Done. Help now reads:
>  -s, --spec=SPEC optionally used with -m or -b"
>                              can be supplied multiple times to build multiple .spec files at once"

---

> I have added the following commit to fix the build of Xperia XA2:
> [sledges@c8b9c55](https://github.com/sledges/droid-hal-device/commit/c8b9c559e9a67ddfab9e8a8e53e0d6508caeca69)
> 
> Full instructions how to build XA2 (including utilising multiple .spec builds) will be made available after this PR is merged (similar to https://sailfishos.org/wiki/Sailfish_X_Build_and_Flash which was only for the Xperia X series)

That's great to hear.

**Comment by ix5** [comment]
Thanks for reviewing and merging!
